### PR TITLE
Inheart 152 - add and use CategoryFilter

### PR DESCRIPTION
**Pull request by pmusialik-intive**
Added CategoryFilter component (select) - not perfect styling yet
Initialized Auctions Page so that I could use CategoryFilter component with some state - passed from the page component
Added data library with sample categories 
Added Select atom 

TODO in the future 
- make categories filter styling perfect,
- make test working (both select and category filter) - currently the same issue (probably) as in Icons (maybe because some icon is used there),
- storybook does not have proper theme? The icon in select should be smaller for example - to investigate 

Zrzut ekranu 2022-10-24 o 13 25 34

### CODE REVIEW

**Comment by konradhanus** [comment]
- [] - add work to do to separate story in jira

File: apps/mobile/src/app/components/Auctions/AuctionsTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <View style={styles.container}>
+      <CategoryFilter
+        items={mapCategoriesToSelect(categories)}
+        selectedValue={selectedCategoryID}
+        onChange={onCategoryChange}
+      />
+    </View>
+  );
+}
+const styles = StyleSheet.create({
```
Comment (konradhanus) [inline apps/mobile/src/app/components/Auctions/AuctionsTemplate.tsx:26]: to consider, styled-components

File: libs/data/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# data
+
+This library was generated with [Nx](https://nx.dev).
```
Comment (konradhanus) [inline libs/data/README.md:3]: add description for what is this library

File: libs/ui/molecules/src/lib/CategoryFilter/__stories__/CategoryFilter.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+import { Column, NativeBaseProvider } from 'native-base';
+import { storiesOf } from '@storybook/react-native';
+import { CategoryFilter } from '../CategoryFilter';
+import { categoriesMock } from '@inheartive/data';
+import { mapCategoriesToSelect } from '../utils/mapCategoriesToSelect';
+
+storiesOf('CategoryFilter', module).add('Basic', () => (
+  <NativeBaseProvider>
```
Comment (konradhanus) [inline libs/ui/molecules/src/lib/CategoryFilter/__stories__/CategoryFilter.stories.tsx:10]: - [ ] - use decorator

Comment (konradhanus) [reply libs/ui/molecules/src/lib/CategoryFilter/__stories__/CategoryFilter.stories.tsx:10]: `.addDecorator((story) => <NativeBaseProvider>{story()}</NativeBaseProvider>)`

File: libs/ui/molecules/src/lib/CategoryFilter/utils/mapCategoriesToSelect.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { ISelectItemProps } from 'native-base';
+import { ICategory } from '@inheartive/data';
+
+export const mapCategoriesToSelect = (categories: ICategory[]): ISelectItemProps[] => {
+  const items = categories.map((category) => {
```
Comment (konradhanus) [inline libs/ui/molecules/src/lib/CategoryFilter/utils/mapCategoriesToSelect.ts:5]: - [ ] - add types

**Review by konradhanus — APPROVED** [review]

**Review by KonradKarlinski — APPROVED** [review]

**Pull request merged**